### ISSUE TITLE
Add Windows 11 support

### ISSUE THREAD

**Issue by thewanderingtraderm** [maharmstone/quibble#105]

**Comment by SzczurekYT** [comment]
I'm not the author, but to my knowledge this will probably going to be added, but currently Quibble can't even support latest windows 10 due to a some change in it, breaking Quibble with no known solution yet.

**Comment by admin-elvistrujanovic** [comment]
Yes - this is also wanted but not only this

We also need an Quiling/Unicorn Framework implemented into this bootloader that will at boot emulate CPU architecture for example to support and execute Windows On ARM versions on Intel/AMD-based PCs with Intel/AMD CPUs

**Comment by maharmstone** [comment]
I think reimplementing Qemu might be a _little_ out of the scope of the project...

**Comment by admin-elvistrujanovic** [comment]
> I think reimplementing Qemu might be a _little_ out of the scope of the project...

Well who will do it if no one who are skilled enough doesn't want to spend time on coding and do this thing... There is literally no such a thing created that will be able to do this - there is no even an method of running another architecture executables dynamically on Windows without VM just like Qemu provided for linux systems...

**Comment by binarymaster** [comment]
> There is literally no such a thing created that will be able to do this - there is no even an method of running another architecture executables dynamically on Windows without VM

Wrong: https://virtuallyfun.com/2024/02/17/announcing-emuwow-beta-1-axp-preview1/

**Comment by admin-elvistrujanovic** [comment]
> > There is literally no such a thing created that will be able to do this - there is no even an method of running another architecture executables dynamically on Windows without VM
> 
> Wrong: https://virtuallyfun.com/2024/02/17/announcing-emuwow-beta-1-axp-preview1/

For a very long time I've searched for something like this and I never found it until now - thanks for sharing the link to that project - I hope that the dev will see there too an suggestion in the already created issue by someone suggesting ARM emulation so maybe we can see some wonders like this in the future : ) - I love this new year 2024 : )

**Comment by maharmstone** [comment]
@binarymaster - thanks, that was an interesting read

@admin-elvistrujanovic - I think the way to approach this would be to add ARM support to Quibble, which I plan to do eventually, and have a separate EFI application acting as a generic emulator. Like I said though, this latter bit would be a pretty gigantic project largely unrelated to everything else (as cool as it would be).

**Comment by BHTY** [comment]
Hi, this is coming a little late but I'm the developer of EmuWoW and I heard through the grapevine that my project had come up here. First of all, I took a brief look at Quibble and it's interesting work, but I have to agree with mahramstone that at least from where I'm sitting, implementing QEMU on bare metal is a little out of the scope of this project, since that's essentially what would be required to boot ARM64 Windows on x64 machines (though obviously support for running Quibble on ARM machines to use it as a bootloader there would be neat). Also, EmuWoW isn't really what you're looking for if that was the goal you were looking to accomplish, for the following reasons

- EmuWoW is for _application_ compatibility, not system emulation - it just emulates the CPU and thunks API calls up to the host, it can't be used to boot an entire Windows install
- Only support for emulating the MIPS and DEC Alpha CPUs is in there right now and is incomplete (it will improve and more CPUs may be added, but that's where it is right now)

**Comment by mateli** [comment]
@joeyoropesa-dev 
> Yes - this is also wanted but not only this
> 
> We also need an Quiling/Unicorn Framework implemented into this bootloader that will at boot emulate CPU architecture for example to support and execute Windows On ARM versions on Intel/AMD-based PCs with Intel/AMD CPUs

Running Windows inside binary translation or an emulator is extremely slow. Running Windows applications inside a binary translator or or on the other hand can be done either by running wine inside qemu or by retrowin32. Neither of those solutions work well quite yet but they are considerably easier starting points than running a full Windows inside qemu. 

Running a full operating systems inside an emulator means that you will spend a large amount of CPU time running basic OS stuff  like scheduling tasks and drawing the desktop (which is most likely going to be software rendered.

So if you have a list if you want to develop something that run foreign windows applications I would suggest that you for starters do this on a native operating system which could be Windows, Linux, FreeBSD or whatever gives you best access to the hardware you are running on. 

Wine is the most mature way to run Windows applications on Linux and FreeBSD but if you want a binary translation layer then this is tricky. The solution here is to run wine inside qemu with the main problem being that this is not at all supported by wine. Usually it is possible and you can get most applications supported by wine working but do not expect this to be a pleasant experience. Wine is not designed for this and to make it work well you will have to make major improvements to how wine work. Also note that currently you will have to run the entirety of wine inside QEMU which is slow. Not as slow as running the entirety of Windows inside an emulator but still very slow.

Retrowin32 on the other hand is a rather fresh project that can run 32 bit x86 applications. Unlike Wine it is an emulator and it is designed to run on windows applications on different hardware. What you would have to do first is to add emulation for the CPU architectures that you need as your applications will obviously not work without this. Then you will have to extend the set of Windows API:s supported by retrowin32 until it runs your applications. Now a major advantage here is that everything that is done inside a Windows API here runs as platform native code and is therefore much faster than running wine inside QEMU.

How difficult this is depends entirely on how difficult your applications are to run within the emulation layer. It is possible that the easiest option are to re-implement the application using a portable framework that can run natively on whatever operating system and hardware you are using. 

@BHTY I noticed that EmuWoW is rather similar to what retrowin32 support. As such I would find it more interesting if you used retrowin32's implementation of the Windows API so that running the MIPS/Alpha binaries are not dependent on Windows.

As for the "rewrite" option minesweeper has a rewrite in winemine (part of wine), gnome mines, kmines and a significant list of other implementations. FreeCell also have numerous alternatives such as AisleRiot. 

Obviously with application level emulation you do not need a bootloader to run your windows applications on BTRFS, you just install Linux on BTRFS and run them using the emulator of your choice. 

Finally running an application using any of the suggested paths is a significant undertaking. However compared to the initial suggestion of running the entirety of Windows inside a binary translator would require a lot more work and run a lot slower.

Also running qemu bare metal do not provide a significant advantage over installing Linux and running qemu even if you would be able to pull it of. Would you run Windows BTRFS drivers inside the emulator? Well then your computer will probably spend most of it's time within BTRFS code.

**Comment by inschrift-spruch-raum** [comment]
I think our discussion should return to the support for Windows 10 22H2 and Windows 11.
So, has the issue been resolved?

**Comment by xproot** [comment]
> I think our discussion should return to the support for Windows 10 22H2 and Windows 11. So, has the issue been resolved?

I attempted to run Windows 24H2 with no success. In fact, I was able to run Windows 10 LTSC 2021 (21H2) but when I tried it on Windows 19 LTSC 2018 (1809) and updated it, it didn't work.

**Comment by inschrift-spruch-raum** [comment]
> I attempted to run Windows 24H2 with no success. In fact, I was able to run Windows 10 LTSC 2021 (21H2) but when I tried it on Windows 19 LTSC 2018 (1809) and updated it, it didn't work.

Do you have any clues now

**Comment by mateli** [comment]
What about trying to boot Windows 11 in Legacy mode? With TPM, Bitlocker etc disabled?

**Comment by inschrift-spruch-raum** [comment]
The root of the problem likely lies elsewhere.

**Comment by mateli** [comment]
I think it's important here to realize what people are trying to achieve. Running an emulator on top of for example Linux can be a perfectly suitable solution. If you want BTRFS and control of the boot process there are a lot of options. For example there are several good PC emulators that can run multiple versions of Windows. There is a severe speed penalty of course but does it matter? It depends on your use case. 

If you want to run special Windows applications then wine or even retrowin32 can be a useful choice. If you a sufficient budget or the ability to spend your own time on it you can extend these solutions to do whatever you want. Most people can of course not do more than check if their applications are supported but in case someone with a late wallet mad skills are reading this then this is something that I would recommend. If you have a windows application and manage to get it running in retrowin32 then you have s solution that will work until the end of time. And there are ways to make emulation as fast as native applications but it of course take development effort.

Finally porting and replacing applications are a useful option. How Firefox and Chrome completely replaced the Windows-only Internet Explorer is an example. Many laptop can now have firmware updates using standardized linux tools. Thats a huge advantage over the Windows way where it's not standardized but requires proprietary software from the manufacturer. 

ReactOS is also a possible path. It's far from a ready to use replacement for Windows. But again if you have resources or mad skills you can take part in fixing this. However I don't think this is the best path forward. The ideas behind retrowin32 require a lot less work.

What I myself is missing from all this options is an open source desktop environment with the look and feel of Window. Not an operating system or compatibility layer but sometimes that look and feels like windows. Obviously with support for the "classic" look. This would make it easier to create Windows-like applications. Something that from a users standpoint are difficult to distinguish from Windows but from a developers perspective can be completely different. Compared to ReactOS it in itself is just an application and doesn't have stuff like device drivers. Where ReactOS doesn't yet have stable USB supposed a solution like this just use libusb or similar. And compared to wine it doesn't actually run Windows applications so a compatibility layer like wine would be needed but not as a core part of the project. Why do I suggest this as superior to any option above? The idea is to minimize development cost by not having to deal with compatibility or reinvent something that for example Linux already can do. As for the standard question "Can it run Photoshop" the answer is "Ask Adobe". That's quite a contrast from Wine. This approach puts the responsibility of running a specific application on the application developer. I would not spend a single hour on getting third party applications running. Until I'm satisfied with every core part of a standard windows desktop being done and the entire desktop environment being stable and useful - that's where to put the effort. Only when that's done I'd start working on replicating Windows applications. Some like notepad would be a necessary part of the desktop environment as it replaces something that shipped with windows. If I started such a project right now I'd do it in C# in order to shorten development time. 

Of course I know that most of this ramble isn't about boot loaders. But people land here for different reasons and I think that it's important to have knowledge about the options.

**Comment by seacremai** [comment]
is there any alternatiev to quibble which works on win 11 with btrfs

**Comment by mateli** [comment]
> is there any alternatiev to quibble which works on win 11 with btrfs

Nope. And for the reasons talked about above there probably never will be. The next version of windows are rumored to only allow signed applications which means you need to either get them from Store or you need your windows attached to an MSDN subscription. This means that secure boot and TPM will be completely required. They sort of already is but tools like Rufus can disable it.

**Comment by seacremai** [comment]
fkin big corpos breaking shit i like

**Comment by maharmstone** [comment]
> is there any alternatiev to quibble which works on win 11 with btrfs

No. I don't anticipate adding Windows 11 support to be a major change, it's just that I've been busy with other things.

> The next version of windows are rumored to only allow signed applications which means you need to either get them from Store or you need your windows attached to an MSDN subscription. This means that secure boot and TPM will be completely required.

This is nonsense.

**Comment by seacremai** [comment]
im hoping for win 11 quibble 🙏

**Comment by mateli** [comment]
They already did it more than once. Secure Boot didn't originally have any other options than using the authorized windows bootloader. While they had to back down from that, there are still hardware being sold that will only boot the authorized windows bootloader.

Then they launched Windows S that will not run any non-authorized software. 

With Windows 11 all the marketing material says that Secure Boot with a TPM is mandatory and S-mode is enabled by default.

Yes there are workarounds to boot windows without Secure Boot/TPM/BitLocker those are still just workarounds and not officially supported behavior. 

The only conclusion one can make is that this lockdown hasn't fully happened yet is to big backlash. But they will absolutely keep pushing for it.

What I would expect is that with Windows 12 consumer laptop's will ship with mandatory windows only secure boot and S-mode while "professional" laptops will not. Then if they fail to sell these locked laptops they'll back down and wait some time before pushing it again. 

The problem of course is that each time they push out something like this the resistance will be weaker. Windows is rapidly losing usage share? And who is leaving? Well mostly people that care about these things. Eventually to few windows users will demand running non-authorized software for Microsoft to back down. And I'm one of those. I don't really need Windows and very rarely use it. Now that there are a growing number of coreboot laptops from linux friendly manufacturers I will probably don't care if we're locked out of secure boot.
